tests/test_backends.py

```diff
@@ -0,0 +1,164 @@
+           'last_name': ''},
+          {'username': ''},
+          )
+    def test_get_user_details(self, **kwargs):
+        """
+        This method is used to verify that the user_details are returned correctly whenever those are or not in the
+        response
+        Expected behavior:
+        Returns a dict with the corresponding information
+        """
+        response = self.cas_backend.get_user_details(kwargs)
+
+        for key, value in kwargs.items():
+            if value:
+                self.assertEqual(value, response[key])
+
+    @unpack
+    @data(
+        (
+            {'username': 'test_has_value'},
+            {}
+        ),
+        (
+            {'username': '123'},
+            {}
+        ),
+        (
+            {'username': 'abc'},
+            {}
+        ),
+    )
+    def test_get_user_id_exists(self, details, response):
+        """
+        This method is used to verify the username is returned when the details include it
+        Expected behavior:
+        Returns the username sended
+        """
+        username = self.cas_backend.get_user_id(details=details, response=response)
+        self.assertEqual(username, details['username'])
+
+    def test_get_user_id_doesnt_exists(self):
+        """
+        This method is used to verify the method throws an error when the username is not defined
+        Expected behavior:
+        Throws an KeyError exception
+        """
+        self.assertRaises(KeyError, lambda: self.cas_backend.get_user_id(details={}, response={}))
+
+    @patch('openedx_cas.backends.get_cas_client')
+    @data(
+        ('edxplatform', [], None)
+    )
+    @unpack
+    def test_cas_validation_returns_username(self, username, attributes, pgtiou, get_cas_client):
+        """
+        This method is used to verify the cas_validation method returns none when there is no username found
+        Expected behavior:
+        Returns none
+        """
+        get_cas_client.return_value.verify_ticket.return_value = (username, attributes, pgtiou)
+        request = Mock()
+        ticket = 'sadasdasd-asdasdasd'
+        service = 'local.overhang.io'
+        response = self.cas_backend.cas_validation(request, ticket, service)
+        self.assertDictEqual({
+            "username": username,
+        },
+            response
+        )
+
+    @patch('openedx_cas.backends.get_cas_client')
+    def test_cas_validation_returns_none_when_no_username(self, get_cas_client):
+        """
+        This method is used to verify the cas_validation method returns the username whenever it's found
+        Expected behavior:
+        Returns a dict with a key called 'username'
+        """
+        get_cas_client.return_value.verify_ticket.return_value = (None, None, None)
+        request = Mock()
+        ticket = 'sadasdasd-asdasdasd'
+        service = 'local.overhang.io'
+        self.assertRaises(AuthFailed, lambda: self.cas_backend.cas_validation(request, ticket, service))
```

Comment:
why lambda? I'm sure unittest has other utilities to do this

Reply:
Yes, we can just pass the arguments